modules/sdk-core/src/bitgo/wallet/wallet.ts

```diff
@@ -1463,14 +1465,111 @@ export class Wallet implements IWallet {
+    if (!params.shareOptions || Object.keys(params.shareOptions).length === 0) {
+      throw new Error('shareOptions cannot be empty');
+    }
+    const bulkCreateShareOptions: BulkCreateShareOptions = { shareOptions: [] };
+
+    for (let index = 0; index < params.shareOptions.length; index++) {
+      const shareOption = params.shareOptions[index];
+
+      common.validateParams(shareOption, ['user', 'permissions'], ['walletPassphrase']);
+
+      if (shareOption.keychain) {
+        const { toPubKey, path } = shareOption.keychain;
+        if (!toPubKey || !path) {
+          throw new Error('Keychain parameters require toPubKey, and path.');
+        }
+      }
+
+      const needsKeychain = shareOption.permissions && shareOption.permissions.indexOf('spend') !== -1; // Remove !shareOption.keychain
+      let sharedKeychain: any = undefined;
+
+      if (needsKeychain) {
+        try {
+          const keychain = await this.getEncryptedUserKeychain();
+
+          if (!keychain.encryptedPrv) {
+            sharedKeychain = undefined;
+          } else {
+            const userPrv = decryptKeychainPrivateKey(this.bitgo, keychain, params.walletPassphrase);
+            if (!userPrv) {
+              throw new Error('Unable to decrypt user keychain.');
+            }
+
+            const eckey = makeRandomKey();
+            const sharing = shareOption;
+            assert(sharing.keychain?.toPubKey);
+
+            const secret = getSharedSecret(eckey, Buffer.from(sharing.keychain?.toPubKey, 'hex')).toString('hex');
+            const newEncryptedPrv = this.bitgo.encrypt({ password: secret, input: userPrv });
+
+            let pub = keychain.pub ?? keychain.commonPub;
+            if (keychain.commonKeychain) {
+              pub =
+                this.baseCoin.getMPCAlgorithm() === 'eddsa'
+                  ? EddsaUtils.getPublicKeyFromCommonKeychain(keychain.commonKeychain)
+                  : EcdsaUtils.getPublicKeyFromCommonKeychain(keychain.commonKeychain);
+            }
+
+            sharedKeychain = {
+              pub,
+              encryptedPrv: newEncryptedPrv,
+              fromPubKey: eckey.publicKey.toString('hex'),
+              toPubKey: sharing.keychain?.toPubKey,
+              path: sharing.keychain?.path,
+            };
+          }
+        } catch (e) {
+          if (e.message === 'No encrypted keychains on this wallet.') {
+            sharedKeychain = {};
+          } else {
+            throw e;
+          }
+        }
+        bulkCreateShareOptions.shareOptions.push({
+          user: shareOption.user,
+          permissions: shareOption.permissions,
+          keychain: Object.keys(sharedKeychain ?? {}).length === 0 ? undefined : sharedKeychain,
+        });
+      }
+    }
+    return await this.createBulkShares(bulkCreateShareOptions);
+  }
+
+  /**
+   * Send an encrypted wallet share to BitGo.
+   * @param params
```

Comment:
?

Reply:
removed

Reply:
no you shouldn't remove them, you should add move documentation on the parameters instead of `params params`

Reply:
and the jsdoc description doesn't state this a /bulk/ key share method.